custom/icds_reports/sqldata/exports/poshan_progress_report.py

```diff
@@ -0,0 +1,218 @@
+            row_data_dict[id[unique_id]] = dummy_row
+        # update the dict
+        # {'unique_id': [contains the excel row with sum of col values for all months eg. m1+m2+m3]}
+        for row in data:
+            row_data = row_data_dict[row[unique_id]][:]
+            for k, v in row.items():
+                if k == 'state_name' or k == 'district_name':
+                    row_data[cols_comprehensive.index(k)] = v
+                elif k != unique_id:
+                    row_data[cols_comprehensive.index(k)] += v if v else 0
+            row_data_dict[row[unique_id]] = row_data
+
+        # calculating average
+        # m1+m2+m3/3
+        total_row = ['Total' for _ in range(0, aggregation_level)] + [0 for _ in range(0, len(
+            headers_comprehensive) - aggregation_level)]
+        for k, v in row_data_dict.items():
+            for col in cols_comprehensive[2:]:
+                if col != 'state_name' and col != 'district_name' and col != unique_id:
+                    val = v[cols_comprehensive.index(col)]
+                    row_data_dict[k][cols_comprehensive.index(col)] = val / 3 if val else 0
+                    total_row[cols_comprehensive.index(col)] += val if val else 0
+
+        row_data_dict["total_row"] = total_row
+
+        # calculating percentage
+        # percent(current col) = 100 * (actual_value(prev col) / expected_value(prev prev col))
+        for k, v in row_data_dict.items():
+            i = 6 + aggregation_level
+            while i < len(cols_comprehensive):
+                row_data_dict[k][i] = 100 * (v[i - 1] / v[i - 2]) if v[i - 2] else 0
+                i = i + 3
+
+        if self.layout != 'comprehensive':
+            headers = headers_summary
+            for k, v in row_data_dict.items():
+                row_data_dict[k] = [v[cols_comprehensive.index(column)] for column in cols_summary]
+
+        excel_rows = [headers]
+        for _, v in row_data_dict.items():
+            excel_rows.append(v)
+        return excel_rows
+
+    def month_wise(self, filters, order_by, aggregation_level):
+        """
+        :param filters: monthwise filter [month=month]
+        :param order_by: order by columns
+        :return: excel_rows
+        """
+        query_set = PoshanProgressReportView.objects.filter(**filters).order_by(*order_by)
+        headers_comprehensive, cols_comprehensive, headers_summary, cols_summary = self.row_constants
+        data = query_set.values(*cols_comprehensive)
+        excel_rows = [headers_comprehensive]
+        total_row = ['Total' for _ in range(0, aggregation_level)] + [0 for _ in range(0, len(
+            headers_comprehensive) - aggregation_level)]
+        for row in data:
+            row_data = []
+            for col in cols_comprehensive:
+                row_data.append(row[col])
+                if col != 'state_name' and col != 'district_name':
+                    total_row[cols_comprehensive.index(col)] += row[col] if row[col] else 0
```

Comment:
whic is this `if` actually checking here?

Reply:
that the value don't some None (like you can't add a Null value) added for safety to avoid exception and errors